resources/js/components/comments.coffee

```diff
@@ -26,6 +26,7 @@ export class Comments extends React.PureComponent
       # TODO: comments should be passed in as props?
       comments = uiState.comments.topLevelCommentIds.map (id) -> store.comments.get(id)
       pinnedComments = uiState.comments.pinnedCommentIds.map (id) -> store.comments.get(id)
+      allPinned = comments.every (comment) => comment.pinned
```

Comment:
```suggestion
      comments = uiState.comments.topLevelCommentIds
        .map (id) -> store.comments.get(id)
        .filter (comment) -> comment? && !comment.pinned
      pinnedComments = uiState.comments.pinnedCommentIds.map (id) -> store.comments.get(id)
```

And then just check for pinnedComments count and remove the `comment.pinned == pinned` filter in the `renderComments`.